microprofile/WFLY-19954_Preview_Support_vertx_feature_pack.adoc

```diff
@@ -0,0 +1,234 @@
+
+The idea of introducing the qualifier is to make CDI lookup side can opt in until this feature pack has high stability level.
+
+The qualifier was selected after discussions with SmallRye team on the related issues above.
+
+==== Get the Vertx instance with the capability name in other subsystems
+The Vertx instance is also a MSC Service so that other services can access it with the capability name:
+[source,java]
+----
+CapabilityServiceBuilder<?> sb = context.getCapabilityServiceTarget().addService();
+Supplier<VertxProxy> vertxProxySupplier = sb.requiresCapability("org.wildfly.extension.vertx", VertxProxy.class);
+----
+NOTE: The Vertx MSC Service is only available when it is set up.
+
+=== Nice-to-Have Requirements
+As the Vertx instance is exposed to the CDI container, it would be good if it can be accessed in the application codes. As the usage of the Vertx instance in application code is undefined, a warning log message can be printed to notify that it is experimental.
+
+Besides configuration each field in the management model, it would be good to configure the {vertx_options_with_link} from a JSON file in the server's configuration directory.
+
+=== Changed requirements
+N/A
+
+=== Non-Requirements
+There is no plan currently of a clustered Vertx instance within WildFly, so some clustering related options will not be configured using `vertx` subsystem, including `EventBusOptions`.
+
+`MetricsOptions` is not required to configure.
+`TracingOptions` is not required to configure.
+
+=== Future Work
+The initial stability level is targeting `Preview`, and the plan is to promote it to higher stability level after some baking time in community.
+
+New attributes maybe added in the future according to the changes in the {vertx_options_with_link}, the vertx subsystem needs to be updated manually to match the new fields.
+
+== Backwards Compatibility
+N/A
+
+=== Default Configuration
+The `vertx` extension/subsystem needs to be activated explicitly to work:
+[source,shell]
+----
+if (outcome != success) of /subsystem=vertx:read-resource
+  echo INFO: Adding vertx subsystem.
+  /extension=org.wildfly.extension.vertx:add
+  /subsystem=vertx:add
+  :reload
+else
+  echo INFO: vertx subsystem already in configuration, subsystem not added.
+end-if
+----
+
+There is no Vertx instance set up by default, users need to do it explicitly like:
+[source,shell]
+----
+/subsystem=vertx/vertx=vertx:add()
+----
+
+=== Importing Existing Configuration
+N/A
+
+=== Deployments
+N/A
+
+=== Interoperability
+N/A
+
+== Implementation Plan
+The {wildfly_vertx_feature_pack_with_link} repository is hosted under wildfly-extras github organization. It contains the source codes for all subsystem functionalities, unit tests and integration tests, including the tests on the vertx subsystem management model operations.
```

Comment:
Is the goal of this effort to move the sources from the external FP into the WF source tree?

Reply:
No, the source code will stay in the external FP.

Reply:
I added a consideration of moving it to WildFly codebase in the future when it becomes mature enough and has a  higher stability level.